libs/sdk-ui-geo/src/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -154,25 +182,40 @@ function renderPushpinColorLegend(props: IGeoChartLegendRendererProps, hasColorL
         geoData,
         position = "top",
         responsive,
-        showFluidLegend,
+        isFluidLegend,
         colorLegendValue,
         numericSymbols = [],
+        contentRect,
     } = props;
 
     const data = geoData!.color!.data;
     const format = geoData!.color!.format;
+    const title = geoData?.size?.name;
 
     const dataWithoutNull = data.filter(isFinite);
     const colorData = generateLegendColorData(dataWithoutNull, colorLegendValue);
-    const isSmall: boolean = Boolean(responsive && showFluidLegend);
+    const width = contentRect?.client?.width;
+    const isAutoPosition = isAutoPositionWithPopup(responsive);
+    const isSmall = isAutoPosition && !shouldRenderLargeColorLegend(width);
+    const shouldRenderTitle = isAutoPosition && title && shouldRenderLegendTitles(width);
+    let size: IHeatmapLegendSize = "large";
```

Comment:
The type here is sketchy, why Heatmap?

Reply:
Renamed: https://github.com/gooddata/gooddata-ui-sdk/pull/1341/commits/f4ffbe150f3cf719e65934062fbf9d7c0d7ff5c2 - the IHeatmapLegendSize was confusing because it is used also in geo chart as you pointed out 👍